bin/load-module-esm-support.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+var fs = require('fs');
+var path = require('path');
+var dirnamePath = path.dirname;
+var extnamePath = path.extname;
+var resolvePath = path.resolve;
+
+var cwd = process.cwd();
+var isModulePackage = getPackageType() === 'module';
+
+function getPackageType() {
+    var dir = __dirname;
+
+    while (dir !== '/' && dir !== process.env.HOME) {
```

Comment:
what does this do on windows?

Reply:
not sure, once i know for sure this implementation will be used i'll set node up on windows and test it out

Reply:
The cross platform check I would use is `dir !== path.dirname(dir)`.

Note I would use `do { ... } while (dir !== path.dirname(dir))` - technically if package.json is in the root directory you would want to look at it.  Additionally I'm not aware of any limitation where node.js will not look at/above the home directory.